mypy/semanal.py

```diff
@@ -5148,8 +5132,9 @@ def in_checked_function(self) -> bool:
         # no regular functions.
         return True
 
+    # TODO(tushar): remove `str` type and `code` property from here
```

Comment:
Plugins might rely on `msg: str`, so no need to delete it.

Reply:
Yeah, you're right. But I think it's a better model in the long term, for plugins to define a set of ErrorMessage objects somewhere in their files, instead of doing inline messages. For the same reasons as why we're doing it in mypy.

Could consider a deprecation cycle and enforcing this on plugins too!

Reply:
> Could consider a deprecation cycle

As far as I know, we never did something like this before.